packages/framework/src/client/scheduler.js

```diff
@@ -79,91 +90,141 @@ export default class Scheduler {
-			// Do nothing, there's already a request in flight.
+	scheduleRequest = (
+		requirement,
+		resourceState,
+		resourceName,
+		operation,
+		data = undefined,
+		now = new Date()
+	) => {
+		const identicalInFlightRequest = find(
+			this.getInFlightRequests( resourceName, operation, now ),
+			( request ) => {
+				return isEqual( request.data, data ) || ( isNil( request.data ) && isNil( data ) );
+			}
+		);
+
+		if ( identicalInFlightRequest ) {
+			// Do nothing, there's already an identical request in flight.
 			return;
 		}
 
-		const existingRequest = this.getScheduledRequest( resourceName, now );
+		const existingRequest = this.getScheduledRequest( resourceName, operation, now );
 		if ( existingRequest ) {
-			existingRequest.addRequirement( requirement, resourceState, now );
+			existingRequest.append( requirement, resourceState, data, now );
 		} else {
-			this.requests.push( new ResourceRequest( resourceName, requirement, resourceState, now ) );
+			this.requests.push( new ResourceRequest( requirement, resourceState, resourceName, operation, data, now ) );
 		}
 		this.updateDelay( now );
 	};
 
+	// TODO: Remove this after mutations are updated to no longer use operations directly.
+	scheduleMutationOperation = (
+		operationName,
+		resourceNames,
+		resourceData,
+		now = new Date()
+	) => {
+		resourceNames.forEach( ( resourceName ) => {
+			const data = resourceData ? resourceData[ resourceName ] : undefined;
+			this.scheduleRequest( {}, {}, resourceName, operationName, data, now );
+		} );
+	}
+
 	/**
-	 * Send a list of requests.
-	 * @param {Array} requests The list of requests to send.
+	 * Send a specific list of requests.
+	 * @param {Array} requests The requests to send.
 	 * @param {Date} now The current time.
-	 * @return {Promise} A promise returned from the fetch call.
+	 * @return {Promise} A promise returned from the operation call.
 	 */
-	sendRequests = ( requests, now ) => {
-		if ( requests.length > 0 ) {
-			const promise = this.fetch( requests.map( request => request.resourceName ) );
-
-			requests.forEach( ( request ) => {
-				request.requested( promise, now );
-			} );
-			return promise;
+	sendRequests = async ( requests, now ) => {
+		// Split the requests up by operation, so we can send one of each operation.
+		const requestsByOperation = groupBy( requests, 'operation' );
+		const promises = [];
+
+		if ( ! isEmpty( requests ) ) {
+			this.dataRequested( requests.map( ( request ) => request.resourceName ) );
 		}
-		// No requests to send.
-		return Promise.resolve();
+
+		Object.keys( requestsByOperation ).forEach( ( operationName ) => {
+			// Send one operation, and associate all requests for it
+			const operationFunc = this.operations[ operationName ];
+			promises.push( sendOperation( operationFunc, requests, this.dataReceived, now ) );
+		} );
+
+		// Return a list of operation promises
+		return await Promise.all( promises );
```

Comment:
Is it a problem if some promises fail and some succeed? The overall `promise` will fail is only one promise fails from the array.

Reply:
Yes, I don't think it can be handled any other way at this point. The return for `sendRequests` isn't very consequential as it's only checked in test code and not when the code runs in the application. resolves and rejections are handled for each promise individually.

Reply:
Ah, yes I see 👍